samples/nrf9160/http_update/modem_delta_update/src/main.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <zephyr.h>
+#include <stdio.h>
+#include <string.h>
+#include <bsd.h>
+#include <modem/bsdlib.h>
+#include <modem/at_cmd.h>
+#include "update.h"
+
+#define FOTA_TEST "FOTA-TEST"
+
+static char version[256];
+
+static bool is_test_firmware(void)
+{
+	static bool version_read;
+	int err;
+	
+	if (!version_read) {
+		err = at_cmd_write("AT+CGMR", version, sizeof(version),
+				NULL);
+		if (err != 0) {
+			printk("Unable to read modem version: %d\n", err);
+			return false;
+		}
+
+		if (strstr(version, CONFIG_SUPPORTED_BASE_VERSION) == NULL) {
+			printk("Unsupported base modem version: %s\n", version);
+			printk("Supported base version (set in prj.conf): %s\n",
+					CONFIG_SUPPORTED_BASE_VERSION);
+			return false;
+		}
+
+		version_read = true;
+	}
+
+	return (strstr(version, FOTA_TEST) != NULL);
+}
+
+static const char * get_host(void)
+{
+	return CONFIG_DOWNLOAD_HOST;
+}
+
+static const char * get_file(void)
+{
+	if (is_test_firmware()) {
+		return CONFIG_DOWNLOAD_FILE_FOTA_TEST_TO_BASE;
+	} else {
+		return CONFIG_DOWNLOAD_FILE_BASE_TO_FOTA_TEST;
+	}
+}
+
+static void finished(void)
+{
+	/* No handling needed */
+}
+
+static bool two_leds(void)
+{
+	return is_test_firmware();
+}
+
+void main(void)
+{
+	int err;
+
+	printk("HTTP delta modem update sample started\n");
+
+	printk("Initializing bsdlib\n");
+#if !defined(CONFIG_BSD_LIBRARY_SYS_INIT)
```

Comment:
References to `bsdlib` were renamed to `libmodem` in the `update.c` file in this PR. Why is it not OK there but fine here?

Reply:
That is wrong, it should be bsdlib.

Its a bit complicated since we are waiting for a refactor of "bsdlib" which renames it to "libmodem", and the final version of this file (which we use for full modem update) have to use the renamed version :)